tests/python-gpu/test_gpu_prediction.py

```diff
@@ -378,3 +378,58 @@ def test_predict_dart(self, n_classes):
 
         copied = cp.array(copied)
         cp.testing.assert_allclose(inplace, copied, atol=1e-6)
+
+    def test_dtypes(self):
+        import cupy as cp
```

Comment:
Please add a `pytestmark` at the top of `test_gpu_prediction.py` to guard against the possibility that cupy isn't installed on the machine.
```
pytestmark = pytest.mark.skipif([cupy doesn't exist])
```

Reply:
Thanks for the review. Added the predicate.

Reply:
The predicate should be added at the module level, since there are other test functions that also depends on cuPy.

Reply:
Added to some other functions.  Some tests don't use cupy, so adding to module level seems overkill.